src/pages/resultsView/coExpression/CoExpressionTab.tsx

```diff
@@ -1,50 +1,108 @@
+    readonly selectedProfileX = remoteData<string|undefined>({
+        await: ()=>[
+            this.xProfileOptions
+        ],
+        invoke: () => {
+            if (!this._selectedProfileX && this.xProfileOptions.result!.length > 0) {
+                return Promise.resolve(this.xProfileOptions.result![0].value);
+            } else {
+                return Promise.resolve(this._selectedProfileX);
+            }
         }
-    }
+    });
 
-    @computed get selectedEntrezGeneId():number | undefined {
-        if (this._selectedEntrezGeneId === undefined && this.props.store.genes.isComplete &&
-                this.props.store.genes.result.length > 0) {
-            return this.props.store.genes.result[0].entrezGeneId;
-        } else {
-            return this._selectedEntrezGeneId;
+    readonly selectedProfileY = remoteData<string|undefined>({
+        await: ()=>[
+            this.yProfileOptions
+        ],
+        invoke: () => {
+            if (!this._selectedProfileY && this.yProfileOptions.result!.length > 0) {
+                return Promise.resolve(this.yProfileOptions.result![0].value);
+            } else {
+                return Promise.resolve(this._selectedProfileY);
+            }
         }
-    }
+    });
+
+    readonly selectedGeneticEntityId = remoteData<string>({
+        await: ()=>[
+            this.props.store.geneticEntities
+        ],
+        invoke: () => {
+            if (!this._selectedGeneticEntityId) {
+                return Promise.resolve(this.props.store.geneticEntities.result![0].geneticEntityId.toString());
+            } else {
+                return Promise.resolve(this._selectedGeneticEntityId);
+            }
+        }
+    });
+
+    readonly isSelectedGeneticEntityAGeneSet = remoteData<boolean>({
+        await: ()=>[
+            this.props.store.geneticEntities
+        ],
+        invoke: () => {
+            for (const geneticEntity of this.props.store.geneticEntities.result!) {
+                if (geneticEntity.geneticEntityType === GeneticEntityType.GENESET && geneticEntity.geneticEntityId.toString() === this._selectedGeneticEntityId) {
+                    return Promise.resolve(true);
+                }
+            }
+            return Promise.resolve(false);
+        }
+    });
+
+    readonly selectedGeneticEntityName = remoteData<string|undefined>({
+        await: ()=>[
+            this.props.store.geneticEntities
+        ],
+        invoke: () => {
+            if (this._selectedGeneticEntityId !== undefined) {
+                for (const geneticEntity of this.props.store.geneticEntities.result!) {
+                    if (this._selectedGeneticEntityId === geneticEntity.geneticEntityId.toString()) {
+                        return Promise.resolve(geneticEntity.geneticEntityName);
+                    }
+                }
+                return Promise.resolve(undefined); //Expected not to reach this code, if so, it will return undefined
```

Comment:
would this indicate something has gone wrong, e.g. with data? if so maybe we should actually throw an error rather than return undefined. @alisman what do you think?

Reply:
Yes, it would mean that the promise `this.props.store.geneticEntities` has failed. I'm not sure if I need to throw an error here or this `return` can be just removed. @adamabeshouse @alisman what's your opinion on this?

Reply:
it seems to me the promise selectedGeneticEntityId already handles the case where there's no _seletedGeneticIdentityId be resolving with the first one, so it would seem that the selected name is just the name property of whatever corresponding entity is return by that promise.   So, this promise seems redundant.   Also, see my comment about that promise above.